health/demo/demo.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/zenoss/zenoss-go-sdk/health"
+	"github.com/zenoss/zenoss-go-sdk/health/log"
+	"github.com/zenoss/zenoss-go-sdk/health/target"
+	"github.com/zenoss/zenoss-go-sdk/health/writer"
+)
+
+const (
+	mercedesTarget = "mercedes.citaro"
+
+	speedMetricID       = "speed"
+	stationsCounterID   = "stations"
+	passengersCounterID = "passengers"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Define health tool configuration
+	config := health.NewConfig()
+	config.CollectionCycle = 2 * time.Second
+
+	// define monitored targets
+	busTarget, err := target.New(
```

Comment:
Is the top-level program expected to define the metrics et al for the health components every time or can the health components register what health info will be sent? Will need the latter to support dynamic health component targets.

Reply:
Covered in ZING-17803